hudi-common/src/main/java/org/apache/hudi/common/model/HoodieCommitMetadata.java

```diff
@@ -500,10 +501,13 @@ public static <T> T fromBytes(byte[] bytes, Class<T> clazz) throws IOException {
       if (bytes.length == 0) {
         return clazz.newInstance();
       }
-      return fromJsonString(
-          fromUTF8Bytes(
-              convertCommitMetadataToJsonBytes(deserializeCommitMetadata(bytes), org.apache.hudi.avro.model.HoodieCommitMetadata.class)),
-          clazz);
+      try {
+        return fromJsonString(fromUTF8Bytes(convertCommitMetadataToJsonBytes(deserializeCommitMetadata(bytes), org.apache.hudi.avro.model.HoodieCommitMetadata.class)), clazz);
+      } catch (Exception e) {
+        // fall back to the alternative method (0.x)
+        LOG.warn("Primary method failed; trying alternative deserialization method.", e);
```

Comment:
We should migrate to @bvaradar 's new SER/DE API instead of catching exceptions: https://github.com/apache/hudi/pull/11923